JsHintTask fails when reporter attribute is not set (Trac #1230)

JsHintTask fails when reporter attribute is not set. This config works flawlessly on Phing 2.12.0 and is broken on 2.13.0 


```text
<jshint haltOnError="true" haltOnWarning="false">
    <fileset refid="src.js" />
</jshint>
```

returns (paths are trimmed)

```text
BUILD FAILED
exception 'BuildException' with message 'build.xml:191:18: build.xml:192:61: Unknown severity "warning"' in local/vendor/phing/phing/classes/phing/tasks/ext/ParallelTask.php:89
Stack trace:
https://www.phing.info/trac/ticket/0 local/vendor/phing/phing/classes/phing/UnknownElement.php(100): ParallelTask->main()
https://www.phing.info/trac/ticket/1 local/vendor/phing/phing/classes/phing/Task.php(283): UnknownElement->main()
https://www.phing.info/trac/ticket/2 local/vendor/phing/phing/classes/phing/Target.php(336): Task->perform()
https://www.phing.info/trac/ticket/3 local/vendor/phing/phing/classes/phing/Target.php(366): Target->main()
https://www.phing.info/trac/ticket/4 local/vendor/phing/phing/classes/phing/Project.php(898): Target->performTasks()
https://www.phing.info/trac/ticket/5 local/vendor/phing/phing/classes/phing/Project.php(868): Project->executeTarget('jshint')
https://www.phing.info/trac/ticket/6 local/vendor/phing/phing/classes/phing/Phing.php(679): Project->executeTargets(Array)
https://www.phing.info/trac/ticket/7 local/vendor/phing/phing/classes/phing/Phing.php(191): Phing->runBuild()
https://www.phing.info/trac/ticket/8 local/vendor/phing/phing/classes/phing/Phing.php(316): Phing::start(Array, NULL)
https://www.phing.info/trac/ticket/9 local/vendor/phing/phing/bin/phing.php(58): Phing::fire(Array)
https://www.phing.info/trac/ticket/10 local/vendor/phing/phing/bin/phing(14): require_once('...')
https://www.phing.info/trac/ticket/11 {main}
```

Migrated from https://www.phing.info/trac/ticket/1230
```json
{
    "status": "new", 
    "changetime": "2016-09-29T08:16:06", 
    "description": "JsHintTask fails when reporter attribute is not set. This config works flawlessly on Phing 2.12.0 and is broken on 2.13.0 \n\n{{{\n<jshint haltOnError=\"true\" haltOnWarning=\"false\">\n    <fileset refid=\"src.js\" />\n</jshint>\n}}}\n\nreturns (paths are trimmed)\n{{{\nBUILD FAILED\nexception 'BuildException' with message 'build.xml:191:18: build.xml:192:61: Unknown severity \"warning\"' in local/vendor/phing/phing/classes/phing/tasks/ext/ParallelTask.php:89\nStack trace:\n#0 local/vendor/phing/phing/classes/phing/UnknownElement.php(100): ParallelTask->main()\n#1 local/vendor/phing/phing/classes/phing/Task.php(283): UnknownElement->main()\n#2 local/vendor/phing/phing/classes/phing/Target.php(336): Task->perform()\n#3 local/vendor/phing/phing/classes/phing/Target.php(366): Target->main()\n#4 local/vendor/phing/phing/classes/phing/Project.php(898): Target->performTasks()\n#5 local/vendor/phing/phing/classes/phing/Project.php(868): Project->executeTarget('jshint')\n#6 local/vendor/phing/phing/classes/phing/Phing.php(679): Project->executeTargets(Array)\n#7 local/vendor/phing/phing/classes/phing/Phing.php(191): Phing->runBuild()\n#8 local/vendor/phing/phing/classes/phing/Phing.php(316): Phing::start(Array, NULL)\n#9 local/vendor/phing/phing/bin/phing.php(58): Phing::fire(Array)\n#10 local/vendor/phing/phing/bin/phing(14): require_once('...')\n#11 {main}\n}}}", 
    "reporter": "anonymous", 
    "cc": "", 
    "resolution": "", 
    "_ts": "1475136966642941", 
    "component": "phing-tasks-ext", 
    "summary": "JsHintTask fails when reporter attribute is not set", 
    "priority": "major", 
    "keywords": "", 
    "version": "2.13.0", 
    "time": "2015-12-08T17:15:03", 
    "milestone": "2.16.0", 
    "owner": "", 
    "type": "defect"
}
```

Trac update at 20151226T10:15:25: phinginfo.20.dimon3000@spamgourmet.com commented:

> Sorry for waiting. I've just attached an example project for reproducing this issue. Parallel task is removed. 
> 
> 
> ```text
> BUILD FAILED
> exception 'BuildException' with message 'build.xml:9:57: Unknown severity "warning"' in vendor/phing/phing/classes/phing/tasks/ext/JsHintTask.php:204
> Stack trace:
> https://www.phing.info/trac/ticket/0 vendor/phing/phing/classes/phing/UnknownElement.php(100): JsHintTask->main()
> https://www.phing.info/trac/ticket/1 vendor/phing/phing/classes/phing/Task.php(283): UnknownElement->main()
> https://www.phing.info/trac/ticket/2 vendor/phing/phing/classes/phing/Target.php(336): Task->perform()
> https://www.phing.info/trac/ticket/3 vendor/phing/phing/classes/phing/Target.php(366): Target->main()
> https://www.phing.info/trac/ticket/4 vendor/phing/phing/classes/phing/Project.php(898): Target->performTasks()
> https://www.phing.info/trac/ticket/5 vendor/phing/phing/classes/phing/Project.php(868): Project->executeTarget('jshint')
> https://www.phing.info/trac/ticket/6 vendor/phing/phing/classes/phing/Phing.php(679): Project->executeTargets(Array)
> https://www.phing.info/trac/ticket/7 vendor/phing/phing/classes/phing/Phing.php(191): Phing->runBuild()
> https://www.phing.info/trac/ticket/8 vendor/phing/phing/classes/phing/Phing.php(316): Phing::start(Array, NULL)
> https://www.phing.info/trac/ticket/9 vendor/phing/phing/bin/phing.php(58): Phing::fire(Array)
> https://www.phing.info/trac/ticket/10 vendor/phing/phing/bin/phing(14): require_once('-cut-...')
> https://www.phing.info/trac/ticket/11 {main}
> ```
> 
> P.S. Relax your anti-spam filter. I had to go through a quest to submit this comment :)
> 
> ```text
> Submission rejected as potential spam
> BotScout says this is spam (Y|MULTI|IP|1|MAIL|1|NAME|0)
> ```

--- a/classes/phing/tasks/ext/JsHintTask.php
+++ b/classes/phing/tasks/ext/JsHintTask.php
@@ -74,7 +74,17 @@ class JsHintTask extends Task
      *
      * @var array
      */
-    private $xmlAttributes;
+    private $xmlAttributes = array(
+        'severity' => array(
+            'error' => 'error',
+            'warning' => 'warning',
+            'info' => 'info'
+        ),
+        'fileError' => 'error',
+        'line' => 'line',
+        'column' => 'column',
+        'message' => 'message',
+    );
 
     /**
      * Path where the the report in Checkstyle format should be saved
@@ -135,25 +145,14 @@ class JsHintTask extends Task
     {
         $this->reporter = $reporter;
 
-        switch ($this->reporter) {
-            case 'jslint':
-                $this->xmlAttributes = array(
-                    'severity' => array('error' => 'E', 'warning' => 'W', 'info' => 'I'),
-                    'fileError' => 'issue',
-                    'line' => 'line',
-                    'column' => 'char',
-                    'message' => 'reason',
-                );
-                break;
-            default:
-                $this->xmlAttributes = array(
-                    'severity' => array('error' => 'error', 'warning' => 'warning', 'info' => 'info'),
-                    'fileError' => 'error',
-                    'line' => 'line',
-                    'column' => 'column',
-                    'message' => 'message',
-                );
-                break;
+        if ($this->reporter === 'jslint') {
+            $this->xmlAttributes = array(
+                'severity' => array('error' => 'E', 'warning' => 'W', 'info' => 'I'),
+                'fileError' => 'issue',
+                'line' => 'line',
+                'column' => 'char',
+                'message' => 'reason',
+            );
         }
     }
 

